framework/src/main/java/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -92,11 +93,19 @@ public static void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
             typeArgs.add(typeArg);
 
             // Add mapping from type parameter to the annotated type argument.
-            typeArgMap.put((TypeVariable) typeElement.getTypeParameters().get(i).asType(), typeArg);
+            // In Java 9, the symbol now contains the type annotation that is present in the source
```

Comment:
1. No need to refer to the 8->9 transition, just document what the key is.
2. That documentation is better with the documentation of field `typeArgMap`.